File: V семестр/Паралельне програмування/NoName/Lab6/Lab6/Function3.cpp
#include "stdafx.h"
#include "Function3.h"

Function3::Function3(int sizeOfArrays)
{
	this->sizeOfArrays = sizeOfArrays;
}


Function3::~Function3()
{
}


void Function3::StartFunction(unsigned int& start) {

	unsigned int myStart = clock();
	
	Methods methods(sizeOfArrays);
	s = methods.fillVectorOfOne();
	p = methods.fillVectorOfOne();

	mo = methods.fillMatrixOfOne();
	mr = methods.fillMatrixOfOne();
	ms = methods.fillMatrixOfOne();

	t = methods.sumOfVectors(methods.matrixToVector(methods.multMatrix(mo, methods.vectorToMatrixOnColumn(p))), methods.matrixToVector(methods.multMatrix(methods.multMatrix(mr, ms), methods.vectorToMatrixOnColumn(s))));

	unsigned int myEnd = clock();
	start = myEnd - myStart;

	#pragma omp critical
	{
		printf("Function 3 started\n");
		methods.putVector(t);
		printf("Function 3 ended\n");
	}
}

File: V семестр/Паралельне програмування/NoName/lab3/Function3.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LabPP3
{
    class Function3
    {
        private int sizeOfArrays;
        private int[] t, p, s;
        private int[,] mo, mr, ms;
        private object block = new object();

        public void StartOfFunction(object arg)
        {
            this.sizeOfArrays = (int)arg;
            Methods methods = new Methods(sizeOfArrays);

            //lock (block)
            //{
            // Console.WriteLine("Start Function3");
            p = methods.FillingOfVector();
            s = methods.FillingOfVector();

            mo = methods.FillingOfMatrix();
            mr = methods.FillingOfMatrix();
            ms = methods.FillingOfMatrix();

            t = methods.SumOfVector(methods.MatrixToVector(methods.MultMatrix(mo, methods.VectorToMatrixOnColumn(p))), methods.MatrixToVector(methods.MultMatrix(methods.MultMatrix(mr, ms), methods.VectorToMatrixOnColumn(s))));

            methods.PutVector(t);

            //Console.WriteLine("End Function3");
            //}
        }
    }
}